File: oikeustulkkirekisteri-service/src/main/java/fi/vm/sade/oikeustulkkirekisteri/service/KoodistoService.java
package fi.vm.sade.oikeustulkkirekisteri.service;

import fi.vm.sade.oikeustulkkirekisteri.service.dto.KoodiDto;
import org.springframework.security.access.prepost.PreAuthorize;

import java.util.List;

import static fi.vm.sade.oikeustulkkirekisteri.service.Constants.PUBLIC;

/**
 * User: tommiratamaa
 * Date: 30.5.2016
 * Time: 14.12
 */
public interface KoodistoService {
    @PreAuthorize(PUBLIC)
    List<KoodiDto> getKielet();

    @PreAuthorize(PUBLIC)
    List<KoodiDto> getMaakunnat();
}

File: oikeustulkkirekisteri-public-ui/src/main/webapp/js/app/service/koodistoService.ts
angular.module('publicRegistryApp').service('KoodistoService', ['$http', ($http) => {
  const root = '/oikeustulkkirekisteri-service/api/public/';

  const getKielet = () => {
    return $http.get(root+'kielet');
  };

  const getMaakunnat = () => {
    return $http.get(root+'maakunnat');
  };

  return {
    getKielet:getKielet,
    getMaakunnat:getMaakunnat
  };
}]);
